### PR TITLE
Added Bottom View of a Binary Tree

### LINKED ISSUE #1101: Adding Binary Tree Bottom View in C++, Java and Python

### PR DESCRIPTION
**Pull request by singh-shreya6**
Fixes #1101 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added Bottom View of a Binary Tree

#### Languages Used:
- C++
- Java
- Python

#### Files Added:
- Binary_Tree_Bottom_View.cpp
- Binary_Tree_Bottom_View.java
- Binary_Tree_Bottom_View.py

### CODE REVIEW

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+# Function to print Bottom View of Binary Tree
+def BottomView(root):
+    if root is None:
+        return
+
+    # initialising variables
+    q = queue.Queue()
+    q.put(root)
+    root.distance = 0;
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.py]: remove semicolon.

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    int distance;
+
+    // asigning horizontal distance to each node of Binary Tree
+    // and replacing nodes of same horizontal distance in a map with
+    // key as distance so as to obtain the Bottom View
+    while (!q.empty()) 
+    {
+        // extract the node at the front of queue
+        Node *temp = q.front();
+        distance = temp->distance;
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+        // make key as distance and data as value for map
+        mp[distance] = temp->data;
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        // remove the extract node from queue
+        q.pop();
+
+        // when left child exists, assign horizontal distance to it,
+        // and push it to the queue
+        if (temp->left != NULL)
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        // make key as distance and data as value for map
+        mp[distance] = temp->data;
+        // remove the extract node from queue
+        q.pop();
+
+        // when left child exists, assign horizontal distance to it,
+        // and push it to the queue
+        if (temp->left != NULL) 
+        {
+            temp->left->distance = distance - 1;
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+            q.push(temp->left);
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+        }
+
+        // when right child exists, assign horizontal distance to it,
+        // and push it to the queue
+        if (temp->right != NULL)
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+        {
+            temp->left->distance = distance - 1;
+            q.push(temp->left);
+        }
+
+        // when right child exists, assign horizontal distance to it,
+        // and push it to the queue
+        if (temp->right != NULL) 
+        {
+            temp->right->distance = distance + 1;
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+            q.push(temp->right);
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+         /     \
+        2       3
+      /   \   /   \
+    4      5 6     7
+         /   \
+        /      \
+       8        9
+    */
+    Node *root = new Node(1);
+    root->left = new Node(2);
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around -> across all files

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+    root->right = new Node(3);
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+    root->left->left = new Node(4);
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+    root->left->right = new Node(5);
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+    root->right->left = new Node(6);
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+    root->right->right = new Node(7);
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+    root->left->right->left = new Node(8);
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 17]
```diff
@@ -0,0 +1,131 @@
...
+    root->left->right->right = new Node(9);
```
Comment (MastersAbh) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: space around ->

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp [hunk 18]
```diff
@@ -0,0 +1,131 @@
...
+            1
+         /     \
+        2       3
+      /   \   /   \
+    4      5 6     7
+         /   \
+        /      \
+       8        9
+
+
```
Comment (somya-kapoor) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.cpp]: Remove line127 only one line gap is sufficient

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.java [hunk 19]
```diff
@@ -0,0 +1,134 @@
...
+            1
+         /     \
+        2       3
+      /   \   /   \
+    4      5 6     7
+         /   \
+        /      \
+       8        9
+
+
```
Comment (somya-kapoor) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.java]: Remove line 130 only one line gap is sufficient

File: Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.py [hunk 20]
```diff
@@ -0,0 +1,111 @@
...
+    Input:
+            1
+         /     \
+        2       3
+      /   \   /   \
+    4      5 6     7
+         /   \
+        /      \
+       8        9
+
```
Comment (somya-kapoor) [inline Binary_Tree_Bottom_View/Binary_Tree_Bottom_View.py]: Remove line 106 only one line gap is sufficient

**Comment by singh-shreya6** [comment]
Done @somya-kapoor 
Please review.

**Pull request merged**